src/pages/events/module-development.md

```diff
@@ -14,50 +14,243 @@ You can programmatically register events using the following methods:
+        "custom_design_from_is_formated": true,
+        "news_from_date_is_formated": true,
+        "news_to_date_is_formated": true,
+        "is_changed_categories": false,
+        "is_changed_websites": false,
+        "force_reindex_eav_required": true
+      }
+    }
+  }
+}
+```
+
+You define an array of fields to transmit in your configuration file. Specify any field within an event's `value` object to ensure it is transmitted to an application. If the field is part of a secondary object, such as `stock_data` in the above example, use the format `<object-name>.field`. For example: `stock_data.product_id`.
+
+After you've registered at least one event, run the [events:generate:module command](./commands.md#generate-a-commerce-module-based-on-a-list-of-subscribed-events) to generate the required plugins.
+
 ### io_events.xml
 
-Create the `<module-root>/etc/io_events.xml` file and define a list of events that should always be emitted. Events listed in this file cannot be disabled with the [`events:unsubscribe` command](./commands.md#unsubscribe-from-a-commerce-event).
+Create the `<module-root>/etc/io_events.xml` file and define a list of events that should always be transmitted. Events listed in this file cannot be disabled with the [`events:unsubscribe` command](./commands.md#unsubscribe-from-a-commerce-event).
 
-The following example registers multiple events.
+The following example registers multiple events. The `<fields>` element defines the contents of each transmitted event.
 
 ```xml
-<?xml version="1.0"?>
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module-commerce-events-client/etc/io_events.xsd">
-   <event name="observer.customer_login" />
-   <event name="observer.customer_logout" />
-   <event name="plugin.magento.catalog.api.category_repository.save" />
-   <event name="plugin.magento.catalog.api.category_repository.save" />
-   <event name="plugin.magento.catalog.api.category_repository.delete" />
-   <event name="plugin.magento.catalog.api.product_repository.save" />
-   <event name="plugin.magento.catalog.api.product_repository.delete" />
-   <event name="plugin.magento.customer.api.customer_repository.save" />
-   <event name="plugin.magento.customer.api.customer_repository.delete" />
-   <event name="plugin.magento.sales.api.order_repository.save" />
-   <event name="plugin.magento.sales.api.order_management.place" />
+    <event name="observer.catalog_product_save_after">
+        <fields>
+            <field name="entity_id" />
+            <field name="sku" />
+            <field name="is_new" />
+        </fields>
+    </event>
+    <event name="plugin.magento.gift_wrapping.model.resource_model.wrapping.save">
+        <fields>
+            <field name="id" />
+            <field name="name" />
+            <field name="status" />
+        </fields>
+    </event>
+    <event name="plugin.magento.sales.api.invoice_item_repository.save">         
```

Comment:
Real fields for this event code, there are more but I think it should be enough for the example.
```
    <event name="plugin.magento.sales.api.invoice_item_repository.save">         
        <fields>
            <field name="entity_id" />
            <field name="parent_id" />
            <field name="base_price" />
            <field name="tax_amount" />
            <field name="base_row_total" />
            <field name="discount_amount" />
            <field name="qty" />
        </fields>
    </event>
```